src/common/com/intellij/plugins/haxe/ide/formatter/HaxeFormattingModel.java

```diff
@@ -36,17 +36,16 @@
   private final FormattingModel myModel;
 
   // Need an abstract method call because of an API change.
-  private final ApiInvoker<TextRange> shiftIndentInsideRangeInvoker;
+  private static final MethodWrapper<TextRange> shiftIndentInsideRangeWrapper =
+          new MethodWrapper<TextRange>(FormattingModel.class, "shiftIndentInsideRange",
+                                    IdeaTarget.IS_VERSION_14_1_5_COMPATIBLE
+                                      ? new Class<?>[] {ASTNode.class, TextRange.class, int.class }
+                                      : new Class<?>[] {TextRange.class, int.class} );
```

Comment:
So, this is how we get around method signature changes.  We are again using reflection to look up the API we want to use with a specific version.  We still get the type safety, but at run-time, which isn't ideal, but it gets us away from maintaining multiple versions of the source  file.